From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/bd81621b04af
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: netfront: make req_prod check properly deal with index wraps
References: bsc#1046610

Using > on bare ring indexes is always wrong, as this will fail when the
indexes wrap. Since the private req_prod_pvt is never behind the shared
req_prod, simply use != here.

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- a/drivers/xen/netfront/netfront.c
+++ b/drivers/xen/netfront/netfront.c
@@ -775,7 +775,7 @@ no_skb:
 
 	/* Is the batch large enough to be worthwhile? */
 	if (i < (np->rx_target/2)) {
-		if (req_prod > np->rx.sring->req_prod)
+		if (req_prod != np->rx.sring->req_prod)
 			goto push;
 		return;
 	}
